src/parse.rs

```diff
@@ -226,11 +229,42 @@ pub fn parse_directory(dir: &Directory) -> std::io::Result<Vec<ModuleComponent>>
 }
 
 fn merge_modules(modules: Vec<ModuleComponent>, lang: Language) -> Vec<ModuleComponent> {
-    match lang {
+    let modules = match lang {
         Language::Cpp => cpp::merge_modules(modules),
         Language::Java => java::merge_modules(modules),
         _ => modules,
+    };
+
+    convert_rpc_and_rest_calls(modules)
+}
+
+fn convert_rpc_and_rest_calls(mut modules: Vec<ModuleComponent>) -> Vec<ModuleComponent> {
+    let modules_view = modules.clone();
+    for module in modules.iter_mut() {
+        let module_view = module.clone();
+        for class in module.classes.iter_mut() {
+            let class_view = class.clone();
+            for method in class.component.methods.iter_mut() {
+                let method_view = method.clone();
+                if let Some(body) = method.body.as_mut() {
+                    body.replace_communication_call(
+                        &modules_view,
+                        &module_view,
+                        Some(&class_view),
+                        &method_view,
+                    );
+                }
+            }
+        }
+        for method in module.component.methods.iter_mut() {
+            let method_view = method.clone();
+            if let Some(body) = method.body.as_mut() {
+                body.replace_communication_call(&modules_view, &module_view, None, &method_view);
```

Comment:
I'm interested how this is editing the structure, since it seems to be acting only on copies of the module/method data and not on the modules/methods themselves.

Reply:
It's replacing the CallExpr itself inside of the method body's AST

Reply:
I think I see what you mean, gotcha.

Reply:
It becomes a bit clearer when you look at this code
```rust
let replace_node = |arg: &mut Expr| {
            if let Some(Node::Expr(replacement)) =
                arg.replace_communication_call(modules, module, callee_class, callee_method)
            {
                *arg = replacement;
            }
        };
```

Reply:
I see, since we have pointers we can just indirect to the actual CallExpr itself and edit it that way. Thank you!